cmd/upgrade/machinepool/cmd.go

```diff
@@ -0,0 +1,248 @@
+			return err
+		}
+		if version == "" {
+			r.Reporter.Infof("No upgrade available for the machine pool '%s' on cluster '%s'", machinePoolID,
+				clusterKey)
+			return nil
+		}
+	}
+	version = ocm.GetRawVersionId(version)
+
+	// build the upgrade policy
+	r.Reporter.Debugf("Building and scheduling the upgrade policy")
+	var upgradePolicy *cmv1.NodePoolUpgradePolicy
+	upgradePolicy, err = buildPolicy(r, cmd, version, machinePoolID, scheduleDate, scheduleTime)
+	if err != nil {
+		return fmt.Errorf("Failed to build schedule upgrade for machine pool %s in cluster '%s': %v",
+			machinePoolID, clusterKey, err)
+	}
+
+	// Ask for confirmation
+	if r.Reporter.IsTerminal() && !confirm.Confirm("upgrade machine pool '%s' to version '%s'", machinePoolID,
+		version) {
+		return nil
+	}
+
+	// schedule the upgrade policy
+	err = r.OCMClient.ScheduleNodePoolUpgrade(cluster.ID(), machinePoolID, upgradePolicy)
+	if err != nil {
+		return fmt.Errorf("Failed to schedule upgrade for machine pool %s in cluster '%s': %v",
+			machinePoolID, clusterKey, err)
+	}
+
+	r.Reporter.Infof("Upgrade successfully scheduled for the machine pool '%s' on cluster '%s'", machinePoolID,
+		clusterKey)
+	return nil
+}
+
+func checkNodePoolExistingScheduledUpgrade(r *rosa.Runtime, cluster *cmv1.Cluster, clusterKey string,
+	nodePoolId string) (*cmv1.NodePool, bool, error) {
+	nodePool, scheduledUpgrade, err := r.OCMClient.GetHypershiftNodePoolUpgrade(cluster.ID(), clusterKey, nodePoolId)
+	if err != nil {
+		return nil, false, err
+	}
+	if scheduledUpgrade != nil {
+		r.Reporter.Warnf("There is already a %s upgrade to version %s on %s",
+			scheduledUpgrade.State().Value(),
+			scheduledUpgrade.Version(),
+			scheduledUpgrade.NextRun().Format("2006-01-02 15:04 MST"),
+		)
+		return nil, true, nil
+	}
+	return nodePool, false, nil
+}
+
+func ComputeNodePoolVersion(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster,
+	nodePool *cmv1.NodePool, version string) (string, error) {
+	channelGroup := cluster.Version().ChannelGroup()
+	filteredVersionList, err := GetAvailableVersion(r, cluster, nodePool)
+	if err != nil {
+		return "", err
+	}
+	// No updates available
+	if len(filteredVersionList) == 0 {
+		return "", nil
+	}
+	if version == "" {
+		// Use as default latest version if we don't specify anything
+		version = filteredVersionList[0]
+	}
+
+	if interactive.Enabled() {
+		version, err = interactive.GetOption(interactive.Input{
+			Question: "Machine pool version",
+			Help:     cmd.Flags().Lookup("version").Usage,
+			Options:  filteredVersionList,
+			Default:  version,
+		})
+		if err != nil {
+			return "", fmt.Errorf("Expected a valid machine pool version from interactive prompt: %s", err)
+		}
+	}
+	// This is called in HyperShift, but we don't want to exclude version which are HCP disabled for node pools
+	// so we pass the relative parameter as false
+	version, err = r.OCMClient.ValidateVersion(version, filteredVersionList, channelGroup, true, false)
```

Comment:
Not a blocker or anything to change, but I wonder if we want to do something here.

Just thinking that the UI will need to use something similar.

Reply:
yes, this is a consequence of https://issues.redhat.com/browse/OCM-2182, which means that the requirement is to check HCP_enabled only for control plane. The only thing I could see to improve this is to explictly mark the parameter as HCP_control_plane_enabled or something, but it's a bit late to change the API, so I don't think we can do that.